applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+
+private:
+    ///@name Private static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+    
+    /**
+    * Creates a new tetrahedra3D10
+    * @return the reference to the new tetrahedra
+    */
+    Tetrahedra3D10<Node<3>> GenerateTetrahedra(ModelPart& this_model_part, std::vector<int>& node_ids) {
+        unsigned int i0 = node_ids[0];
+        unsigned int i1 = node_ids[1];
+        unsigned int i2 = node_ids[2];
+        unsigned int i3 = node_ids[3];
+        unsigned int i4 = node_ids[4];
+        unsigned int i6 = node_ids[5];
+        unsigned int i7 = node_ids[6];
+        unsigned int i5 = node_ids[7];
+        unsigned int i8 = node_ids[8];
+        unsigned int i9 = node_ids[9];
+
+        Tetrahedra3D10<Node < 3 > > geom(
+            this_model_part.Nodes()(i0),
+            this_model_part.Nodes()(i1),
+            this_model_part.Nodes()(i2),
+            this_model_part.Nodes()(i3),
+            this_model_part.Nodes()(i4),
+            this_model_part.Nodes()(i5),
+            this_model_part.Nodes()(i6),
+            this_model_part.Nodes()(i7),
+            this_model_part.Nodes()(i8),
+            this_model_part.Nodes()(i9)
+        );
+        return geom;
+    }
+
+    /**
+    * Creates a new triangle3D6
+    * @return the reference to the new triangle
+    */
+    Triangle3D6<Node<3>> GenerateTriangle3D6(ModelPart& this_model_part, array_1d<int, 6>& node_ids) {
+        unsigned int i0   = node_ids[0];
+        unsigned int i1   = node_ids[1];
+        unsigned int i2   = node_ids[2];
+        unsigned int i3   = node_ids[3];
+        unsigned int i4   = node_ids[4];
+        unsigned int i5   = node_ids[5];
+
+        Triangle3D6<Node<3> > geom(
+                this_model_part.Nodes()(i0),
+                this_model_part.Nodes()(i1),
+                this_model_part.Nodes()(i2),
+                this_model_part.Nodes()(i3),
+                this_model_part.Nodes()(i4),
+                this_model_part.Nodes()(i5)
+                );
+        return geom;
+    }
+
+        /**
+    * It erases the old Tetrahedra3D4 elements and it creates the new Tetrahedra3D10 ones
+    * @param Coord: The compressed matrix containing at (i,j) the id of the node created between nodes i,j
+    * @param New_Elements: The new elements created
+    * @param interpolate_internal_variables: A boolean that defines if it is necessary to interpolate the internal variables
+    * @return this_model_part: The model part of the model (it is the input too)
+    */
+
+    void EraseOldElementAndCreateNewElement(
+            ModelPart& this_model_part,
+            const compressed_matrix<int>& Coord,
+            PointerVector< Element >& NewElements,
+            bool interpolate_internal_variables
+    ) override
+    {
+	ElementsArrayType& rElements = this_model_part.Elements();
```

Comment:
```suggestion
	auto& r_elements = this_model_part.Elements();
```
(also note the style guide suggestion, snake_case for local variables).